src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+      signedUnbondingSlashingTx,
+    );
+    if (!unbondingSignatures) {
+      throw new Error("No signature found in the unbonding output slashing PSBT");
+    }
+
+    // Create Proof of Possession
+    const bech32AddressHex = uint8ArrayToHex(fromBech32(bech32Address).data);
+    const signedBbnAddress = await this.btcProvider.signMessage(
+      bech32AddressHex,
+      "ecdsa",
+    );
+    const ecdsaSig = Uint8Array.from(Buffer.from(signedBbnAddress, "base64"));
+    const proofOfPossession: ProofOfPossessionBTC = {
+      btcSigType: BTCSigType.ECDSA,
+      btcSig: ecdsaSig,
+    };
+
+    // Prepare and send protobuf message
+    const msg: btcstakingtx.MsgCreateBTCDelegation =
+      btcstakingtx.MsgCreateBTCDelegation.fromPartial({
+        stakerAddr: bech32Address,
+        pop: proofOfPossession,
+        btcPk: Uint8Array.from(
+          Buffer.from(stakerInfo.publicKeyNoCoordHex, "hex"),
+        ),
+        fpBtcPkList: [
+          Uint8Array.from(
+            Buffer.from(stakingInput.finalityProviderPkNoCoordHex, "hex"),
+          ),
+        ],
+        stakingTime: stakingInput.stakingTimelock,
+        stakingValue: stakingInput.stakingAmountSat,
+        stakingTx: Uint8Array.from(stakingTx.toBuffer()),
+        slashingTx: Uint8Array.from(
+          Buffer.from(clearTxSignatures(signedSlashingTx).toHex(), "hex"),
+        ),
+        delegatorSlashingSig: Uint8Array.from(slashingSig),
+        unbondingTime: param.unbondingTime,
+        unbondingTx: Uint8Array.from(unbondingTx.toBuffer()),
+        unbondingValue: stakingInput.stakingAmountSat - param.unbondingFeeSat,
+        unbondingSlashingTx: Uint8Array.from(
+          Buffer.from(
+            clearTxSignatures(signedUnbondingSlashingTx).toHex(),
+            "hex",
+          ),
+        ),
+        delegatorUnbondingSlashingSig: Uint8Array.from(unbondingSignatures),
+        stakingTxInclusionProof: inclusionProof,
+      });
+
+    return {
+      typeUrl: BBN_REGISTRY_TYPE_URLS.MsgCreateBTCDelegation,
+      value: msg,
+    };
+  };
+
+  private getInclusionProof(
+    inclusionProof: InclusionProof,
+  ): btcstaking.InclusionProof {
+    const {
+      pos,
+      merkle,
+      blockHash
+    } = inclusionProof;
+    const proofHex = deriveMerkleProof(merkle);
+  
+    const hash = reverseBuffer(Uint8Array.from(Buffer.from(blockHash, "hex")));
+    const inclusionProofKey: btccheckpoint.TransactionKey =
+      btccheckpoint.TransactionKey.fromPartial({
+        index: pos,
+        hash,
+      });
+    return btcstaking.InclusionProof.fromPartial({
+      key: inclusionProofKey,
+      proof: Uint8Array.from(Buffer.from(proofHex, "hex")),
+    });
+  };
+}
+
+/**
+ * Extracts the first valid Schnorr signature from a signed transaction.
+ * @param singedTransaction - The signed transaction.
+ * @returns The first valid Schnorr signature or undefined if no valid signature is found.
+ */
+export const extractSchnorrSignaturesFromTransaction = (
```

Comment:
does this function need to be public ? It seems it is mostly a helper to extract sigantures from slashing transaction.

Asking as it is a bit confusing as a whole:
- the name suggests extracting multiple signatures, while docs explain extracting only one
- doc mentions word `valid` , though no validity check is done on signature